pyo3-macros-backend/src/method.rs

```diff
@@ -616,14 +643,10 @@ impl<'a> FnSpec<'a> {
                     .signature
                     .arguments
                     .iter()
-                    .map(|arg| {
-                        if arg.py {
-                            quote!(py)
-                        } else if arg.is_cancel_handle {
-                            quote!(__cancel_handle)
-                        } else {
-                            unreachable!()
-                        }
+                    .map(|arg| match arg.kind {
+                        FnArgKind::Py => quote!(py),
+                        FnArgKind::CancelHandle { .. } => quote!(__cancel_handle),
+                        _ => unreachable!(),
```

Comment:
maybe some kind of 
```rust
enum PyOrCancelHandle{
    Py,
    CancelHandle,
}
```
 could get rid of this unreachable 🤔 Just some food for thoughts

Reply:
Hmm, since this from the general parsed types, I don't think we can do a lot to avoid it. But since this is under `CallingConvention::Noargs` there really shouldn't be any "real" argument's (reaching Python), except for `self` which is handled separately. So I think `unreachable!()` is appropriate here, but an explanation comment might be good, I'll add that.